router/broker.go

```diff
@@ -33,12 +35,38 @@ type subscription struct {
 	subscribers map[*wamp.Session]struct{}
 }
 
+// storedEvent is a wrapper around wamp event message with timestamp
+// to be used in event history
+type storedEvent struct {
+	timestamp    time.Time
+	Subscription wamp.ID
+	Publication  wamp.ID
+	Details      wamp.Dict
+	Arguments    wamp.List `wamp:"omitempty"`
+	ArgumentsKw  wamp.Dict `wamp:"omitempty"`
+}
+
+type eventHistoryItem struct {
+	publication *wamp.Publish
+	event       *storedEvent
+}
+
+type eventHistoryStoreItem struct {
```

Comment:
If this stores event history then the proper name would be `eventHistoryItemStore` but I'd just call it `historyStore` and avoid the long name.